build.sbt

```diff
@@ -334,98 +347,101 @@ lazy val chisel = (project in file("."))
+        "-doc-source-url", {
+          val branch =
+            if (version.value.endsWith("-SNAPSHOT")) {
+              "master"
+            } else {
+              s"v${version.value}"
+            }
+          s"https://github.com/chipsalliance/chisel/tree/$branch€{FILE_PATH_EXT}#L€{FILE_LINE}"
+        },
+        "-language:implicitConversions"
+      ) ++
+        // Suppress compiler plugin for source files in core
+        // We don't need this in regular compile because we just don't add the chisel-plugin to core's scalacOptions
+        // This works around an issue where unidoc uses the exact same arguments for all source files.
+        // This is probably fundamental to how ScalaDoc works so there may be no solution other than this workaround.
+        // See https://github.com/sbt/sbt-unidoc/issues/107
+        (core / Compile / sources).value.map("-P:chiselplugin:INTERNALskipFile:" + _)
+        ++ {
+          CrossVersion.partialVersion(scalaVersion.value) match {
+            case Some((2, n)) if n >= 13 => "-implicits" :: Nil
+            case _                       => Nil
           }
-        s"https://github.com/chipsalliance/chisel/tree/$branch€{FILE_PATH_EXT}#L€{FILE_LINE}"
-      },
-      "-language:implicitConversions"
-    ) ++
-      // Suppress compiler plugin for source files in core
-      // We don't need this in regular compile because we just don't add the chisel-plugin to core's scalacOptions
-      // This works around an issue where unidoc uses the exact same arguments for all source files.
-      // This is probably fundamental to how ScalaDoc works so there may be no solution other than this workaround.
-      // See https://github.com/sbt/sbt-unidoc/issues/107
-      (core / Compile / sources).value.map("-P:chiselplugin:INTERNALskipFile:" + _)
-      ++ {
-        CrossVersion.partialVersion(scalaVersion.value) match {
-          case Some((2, n)) if n >= 13 => "-implicits" :: Nil
-          case _                       => Nil
         }
-      }
-  )
+    )
 
 // End-to-end tests that check the functionality of the emitted design with simulation
- lazy val integrationTests = (project in file("integration-tests"))
-   .dependsOn(chisel % "compile->compile;test->test")
-   .dependsOn(firrtl) // SBT doesn't seem to be propagating transitive library dependencies...
-   .dependsOn(standardLibrary)
-   .settings(commonSettings: _*)
-   .settings(warningSuppression: _*)
-   .settings(fatalWarningsSettings: _*)
-   .settings(chiselSettings: _*)
-   .settings(usePluginSettings: _*)
+lazy val integrationTests = (project in file("integration-tests"))
+  .dependsOn(chisel % "compile->compile;test->test")
+  .dependsOn(firrtl) // SBT doesn't seem to be propagating transitive library dependencies...
+  .dependsOn(standardLibrary)
+  .settings(commonSettings: _*)
+  .settings(warningSuppression: _*)
+  .settings(fatalWarningsSettings: _*)
+  .settings(chiselSettings: _*)
+  .settings(usePluginSettings: _*)
```

Comment:
if we *are* going to format this file in this PR, can we make sure that the process is repeatable. What settings was vscode picking up? I guess I'd strongly prefer we make the "format the build.sbt file and make sure it stays formatted" in its own PR

Reply:
I agree we should make this repeatable, looking in to how we can apply it to the build.sbt

Reply:
There is a command `scalafmtSbt` which will format `*.sbt` files and `project/*.scala` files. We should see if we can make this part of "all". I will say, we should do this as a separate PR rather than as part of this one just to keep the history more clear.